scripts/xtensa-build-zephyr.sh

```diff
@@ -0,0 +1,135 @@
+
+# stop on most errors
+set -e
+
+SUPPORTED_PLATFORMS=(apl cnl icl tgl-h)
+BUILD_PATH=zephyr
+BUILD_ROM=no
+BUILD_JOBS=$(nproc --all)
+RIMAGE_KEY=modules/audio/sof/keys/otc_private_key.pem
+PLATFORMS=()
+
+PATH=$pwd/local/bin:$PATH
+
+pwd=$(pwd)
+
+die()
+{
+	>&2 printf '%s ERROR: ' "$0"
+	# We want die() to be usable exactly like printf
+	# shellcheck disable=SC2059
+	>&2 printf "$@"
+	exit 1
+}
+
+print_usage()
+{
+    cat <<EOF
+Re-configures and re-builds SOF with Zephyr using a pre-installed Zephyr toolchain and
+platform's _defconfig file.
+
+usage: $0 [options] platform(s)
+
+       -a Build all platforms
+       -j n Set number of make build jobs. Jobs=#cores by default.
+           Infinte when not specified.
+       -k Path a non-default rimage signing key.
+       -p Build path.
+
+Supported platforms ${SUPPORTED_PLATFORMS[*]}
+
+EOF
+}
+
+# parse the args
+while getopts "adj:k:p:ruv" OPTION; do
+        case "$OPTION" in
+		a) PLATFORMS=("${SUPPORTED_PLATFORMS[@]}") ;;
+		j) BUILD_JOBS=$OPTARG ;;
+		k) RIMAGE_KEY=$OPTARG ;;
+		p) BUILD_PATH=$OPTARG ;;
+		r) BUILD_ROM=yes ;;
+		*) print_usage; exit 1 ;;
+        esac
+done
+shift $((OPTIND-1))
+
+# parse platform args
+for arg in "$@"; do
+	platform=none
+	for i in "${SUPPORTED_PLATFORMS[@]}"; do
+		if [ x"$i" = x"$arg" ]; then
+			PLATFORMS=("${PLATFORMS[@]}" "$i")
+			platform=$i
+			shift || true
+			break
+		fi
+	done
+	if [ "$platform" == "none" ]; then
+		echo "Error: Unknown platform specified: $arg"
+		echo "Supported platforms are: ${SUPPORTED_PLATFORMS[*]}"
+		exit 1
+	fi
+done
+
+# check target platform(s) have been passed in
+if [ ${#PLATFORMS[@]} -eq 0 ];
+then
+	echo "Error: No platforms specified. Supported are: " \
+		"${SUPPORTED_PLATFORMS[*]}"
+	print_usage
+	exit 1
+fi
+
+# Check out Zephyr + SOF
+
+which west > /dev/null 2>&1 || die "Install west and a west toolchain following https://docs.zephyrproject.org/latest/getting_started/index.html"
+which git > /dev/null 2>&1 || die "Install git"
+
+[ -e $BUILD_PATH ] && die "$BUILD_PATH already exists"
+mkdir -p $BUILD_PATH
+cd $BUILD_PATH
+west init
+west update
+cd modules/audio/sof
+git remote add sof https://github.com/thesofproject/sof.git
+git remote update sof
+git checkout sof/main
+git clone --recurse-submodules https://github.com/thesofproject/rimage.git
+cd -
+
+# Build rimage
+RIMAGE_DIR=build-rimage
+mkdir $RIMAGE_DIR
+cd $RIMAGE_DIR
+cmake ../modules/audio/sof/rimage
+make -j$BUILD_JOBS
+cd -
+
+for platform in "${PLATFORMS[@]}"; do
+	case $platform in
+	apl)
+		PLAT_CONFIG='intel_adsp_cavs15'
+		;;
+	cnl)
+		PLAT_CONFIG='intel_adsp_cavs18'
+		;;
+	icl)
+		PLAT_CONFIG='intel_adsp_cavs20'
+		;;
+	tgl-h)
+		PLAT_CONFIG='intel_adsp_cavs25'
+		;;
+	*)
+		echo "Unsupported platform: $platform"
+		exit 1
+		;;
+	esac
+
+	west build -d build-$platform -p -b $PLAT_CONFIG zephyr/samples/subsys/audio/sof
+	west sign -d build-$platform -t rimage -p $RIMAGE_DIR/rimage \
```

Comment:
I believe for sign, some of these arguments should be optional.
But that should be made from [sign west script](https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/west_commands/sign.py) first and then will update this one.

I'm saying this because imx doesn't need a RIMAGE_KEY key or a [bootloader](https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/west_commands/sign.py#L433).

So, for now, this looks good.